src/com/xilinx/rapidwright/interchange/DeviceResourcesWriter.java

```diff
@@ -348,7 +349,11 @@ public static void writeDeviceResourcesFile(String part, Device device, CodePerf
 
         Netlist.Builder netlistBuilder = devBuilder.getPrimLibs();
         netlistBuilder.setName(netlist.getName());
-        LogNetlistWriter writer = new LogNetlistWriter(allStrings);
+        LogNetlistWriter writer = new LogNetlistWriter(allStrings, new HashMap<String, String>() {{
+                    put(EDIFTools.EDIF_LIBRARY_HDI_PRIMITIVES_NAME, "primitives");
+                    put(device.getSeries()+"_"+EDIFTools.MACRO_PRIMITIVES_LIB, "macros");
```

Comment:
@gatecat Perhaps the `"macros"` and `"primitives"` strings should be declared as a `public static final String` at the top of one of the LogNetlist classes and named such that they indicate a standardized Interchange library name.

Reply:
Good idea, updated